### PR TITLE
Added cloudi.2.0.6

### PR DESCRIPTION
**Pull request by okeuday**

### CODE REVIEW

File: packages/cloudi/cloudi.2.0.6/opam [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+dev-repo: "git+https://github.com/CloudI/cloudi_api_ocaml.git"
+build: [
+  [make]
+]
+depends: [
+  "ocaml" {>= "4.03.0"}
+  "base-unix"
+  "zarith" {>= "1.12"}
+]
+synopsis: "OCaml CloudI API"
```
Comment (mseri) [inline packages/cloudi/cloudi.2.0.6/opam:16]:
```suggestion
conflicts: [ "ocaml-option-bytecode-only" ]
synopsis: "OCaml CloudI API"
```

Comment (mseri) [reply packages/cloudi/cloudi.2.0.6/opam:16]: Due to
```

#=== ERROR while compiling cloudi.2.0.6 =======================================#
# context              2.2.0~alpha~dev | linux/arm32 | ocaml-base-compiler.5.0.0 | pinned(https://github.com/CloudI/cloudi_api_ocaml/archive/v2.0.6.tar.gz)
# path                 ~/.opam/5.0/.opam-switch/build/cloudi.2.0.6
# command              ~/.opam/opam-init/hooks/sandbox.sh build make
# exit-code            2
# env-file             ~/.opam/log/cloudi-7-59cbe0.env
# output-file          ~/.opam/log/cloudi-7-59cbe0.out
### output ###
# (cd external/zarith-1.12 && \
#      ./configure && \
#      make && \
#      cp big_int_Z.cmi big_int_Z.cmx libzarith.a q.cmi q.cmx zarith.a zarith.cma zarith.cmxa zarith_top.cma zarith_top.cmi zarith_version.cmi zarith_version.cmx z.cmi z.cmx ../..)
# binary ocaml: found in /home/opam/.opam/5.0/bin
# binary ocamlc: found in /home/opam/.opam/5.0/bin
# binary ocamldep: found in /home/opam/.opam/5.0/bin
# binary ocamlmklib: found in /home/opam/.opam/5.0/bin
# binary ocamldoc: found in /home/opam/.opam/5.0/bin
# binary gcc: found in /usr/bin
# binary ocamlopt: not found
# checking compilation with gcc -O3 -Wall -Wextra : working
# include caml/mlvalues.h: found
# binary ocamlfind: found in /home/opam/.opam/5.0/bin
# OCaml's word size is 32
# binary uname: found in /bin
# include gmp.h: found
# library gmp: found
# OCaml supports -bin-annot to produce documentation
# 
# detected configuration:
# 
#   native-code:          no
#   dynamic linking:      no
#   defines:              -DHAS_GMP 
#   libraries:             -lgmp
#   C options:            -O3 -Wall -Wextra 
#   installation path:    /home/opam/.opam/5.0/lib
#   installation method   findlib
# 
# configuration successful!
# now type "make" to build
# then type "make install" or "sudo make install" to install
# make[1]: Entering directory '/home/opam/.opam/5.0/.opam-switch/build/cloudi.2.0.6/external/zarith-1.12'
# (echo "let"; grep "version" META | head -1) > zarith_version.ml
# ocamldep -native  zarith_version.ml z.ml q.ml big_int_Z.ml z.mli q.mli big_int_Z.mli > depend
# ocamlc -I +compiler-libs -bin-annot  -c zarith_version.ml
# ocamlc -I +compiler-libs -bin-annot  -c z.mli
# ocamlc -I +compiler-libs -bin-annot  -c z.ml
# ocamlc -I +compiler-libs -bin-annot  -c q.mli
# ocamlc -I +compiler-libs -bin-annot  -c q.ml
# File "q.ml", line 537, characters 9-23:
# 537 |         (Z.of_substring s 0 i)
#                ^^^^^^^^^^^^^^
# Warning 6 [labels-omitted]: labels pos, len were omitted in the application of this function.
# File "q.ml", line 538, characters 9-23:
# 538 |         (Z.of_substring s (i+1) (String.length s-i-1))
#                ^^^^^^^^^^^^^^
# Warning 6 [labels-omitted]: labels pos, len were omitted in the application of this function.
# ocamlc -I +compiler-libs -bin-annot  -c big_int_Z.mli
# ocamlc -I +compiler-libs -bin-annot  -c big_int_Z.ml
# ocamlmklib -failsafe -o zarith zarith_version.cmo z.cmo q.cmo big_int_Z.cmo -lgmp
# ocamlc -ccopt "-I/home/opam/.opam/5.0/lib/ocaml  -DHAS_GMP  -O3 -Wall -Wextra " -c caml_z.c
# ocamlmklib -failsafe -o zarith caml_z.o -lgmp
# ocamlc -I +compiler-libs -bin-annot  -c zarith_top.ml
# ocamlc -o zarith_top.cma -a zarith_top.cmo
# make[1]: Leaving directory '/home/opam/.opam/5.0/.opam-switch/build/cloudi.2.0.6/external/zarith-1.12'
# cp: cannot stat 'big_int_Z.cmx': No such file or directory
# cp: cannot stat 'q.cmx': No such file or directory
# cp: cannot stat 'zarith.a': No such file or directory
# cp: cannot stat 'zarith.cmxa': No such file or directory
# cp: cannot stat 'zarith_version.cmx': No such file or directory
# cp: cannot stat 'z.cmx': No such file or directory
# make: *** [makefile:48: dependency_zarith] Error 1
```

**Comment by mseri** [comment]
The build fails on macos arm64 due to the different include path needed for gmp:
```
#=== ERROR while compiling cloudi.2.0.6 =======================================#
# context              2.1.4 | macos/arm64 | ocaml-base-compiler.4.14.1 | pinned(https://github.com/CloudI/cloudi_api_ocaml/archive/v2.0.6.tar.gz)
# path                 ~/.opam/4.14.1/.opam-switch/build/cloudi.2.0.6
# command              ~/.opam/opam-init/hooks/sandbox.sh build make
# exit-code            2
# env-file             ~/.opam/log/cloudi-71901-cd6072.env
# output-file          ~/.opam/log/cloudi-71901-cd6072.out
### output ###
# (cd external/zarith-1.12 && \
#      ./configure && \
#      /Library/Developer/CommandLineTools/usr/bin/make && \
#      cp big_int_Z.cmi big_int_Z.cmx libzarith.a q.cmi q.cmx zarith.a zarith.cma zarith.cmxa zarith_top.cma zarith_top.cmi zarith_version.cmi zarith_version.cmx z.cmi z.cmx ../..)
# binary ocaml: found in /Users/mac1000/.opam/4.14.1/bin
# binary ocamlc: found in /Users/mac1000/.opam/4.14.1/bin
# binary ocamldep: found in /Users/mac1000/.opam/4.14.1/bin
# binary ocamlmklib: found in /Users/mac1000/.opam/4.14.1/bin
# binary ocamldoc: found in /Users/mac1000/.opam/4.14.1/bin
# binary gcc: found in /usr/bin
# binary ocamlopt: found in /Users/mac1000/.opam/4.14.1/bin
# checking compilation with gcc -O3 -Wall -Wextra : working
# include caml/mlvalues.h: found
# library dynlink.cmxa: found
# binary ocamlfind: found in /Users/mac1000/.opam/4.14.1/bin
# OCaml's word size is 64
# binary uname: found in /usr/bin
# include gmp.h: not found
# include mpir.h: not found
# cannot find GMP nor MPIR
# make: *** [dependency_zarith] Error 2
```
I suggest to use pkg-config in the future

**Comment by okeuday** [comment]
@mseri Thank you for adding `conflicts: [ "ocaml-option-bytecode-only" ]`.  I will look into using `pkg-config` to avoid this problem in the future.  Previously, `num` was being used instead of `zarith` and the `zarith` build system doesn't appear to be as well used (i.e., configure/Makefile is custom and brittle).  It was simpler when OCaml had big integer support included (previously with `num`), but I understand that is unlikely to occur in the future.

**Comment by mseri** [comment]
Future versions of zarith will use pkg-config as well, ao this will no longer be a problem hopefully

**Pull request merged**